### DISCUSSION TITLE
Cannot Switch Chat Box in one page without refreshing

### DISCUSSION THREAD

**Discussion by heatonmar** [chatwoot/chatwoot#9732] [Feature Requests]
### Is your feature or enhancement related to a problem? Please describe.

My website shows namecards of agents, with "Start Chat" button under each agent's namecard. The button was binded by the website token (Each Agent has his own inbox with single inbox_id) of each agent's widget. So when I tried to talk with Agent A, it worked. But when I tried to talk with Agent B, I clicked on start chat button under Agent B's name card. Instead of switching channel, the popout widget still remained on Agent A's window so I found it hard to talk with Agent B (Honestly what I could do was just to refresh the page and start the chat again, which would work but inconvenient to users ). Here's my widget js code:

function initializeChatwoot(contactId, email, websiteToken) {
        alert('Chat Started');
        window.chatwootSettings = {"position":"right","type":"standard","launcherTitle":"Contact Me"};
        (function(d,t) {
          var BASE_URL="http://myserverip:3000";
          var g=d.createElement(t),s=d.getElementsByTagName(t)[0];
          g.src=BASE_URL+"/packs/js/sdk.js";
          g.defer = true;
          g.async = true;
          s.parentNode.insertBefore(g,s);
          g.onload=function(){
            window.chatwootSDK.run({
              websiteToken: websiteToken,
              baseUrl: BASE_URL
            });
 
            window.addEventListener("chatwoot:ready", function () {

                window.$chatwoot.setUser(contactId, {
                    email: email
                });
            });
          }
        })(document,"script");
    }

### Describe the solution you'd like

I'll be appreciated if channel switch could be done without any refreshment

**Comment by pranavrajs** [comment]
@heatonmar We don't support this type of implementation. Can you please share more about the usecase?
